vulnerability/events_creator.go

```diff
@@ -210,23 +211,61 @@ func getIdentifierType(id string) string {
 	return s[0]
 }
 
-func getCVSSScore(vul trivy_types.DetectedVulnerability, source db_types.SourceID) float64 {
-	if cvss, ok := vul.CVSS[source]; ok {
-		return cvss.V3Score
+func (e EventsCreator) getCVSSVersion(vul trivy_types.DetectedVulnerability) string {
+	v := getCVSSVector(vul)
+	if v == "" {
+		e.log.Warnf("No CVSS vector found for vulnerability: %s", vul.VulnerabilityID)
+		return ""
+	}
+
+	parts := strings.Split(v, "/")
+	for _, part := range parts {
+		if strings.HasPrefix(part, vectorHeader) {
+			// Extract the number after "CVSS:"
+			version := strings.TrimPrefix(part, vectorHeader)
+			return version
+		}
+	}
+
+	e.log.Warnf("no CVSS version found in vector: %s", v)
+	return ""
```

Comment:
In case no vector we'll omit the version field, are we fine with it?

Reply:
I've opened a ticket exactly to discuss such behaviors.
- https://github.com/elastic/cloudbeat/issues/870
We can push and then change after consensus between the teams.